analysis/src/lib.rs

```diff
@@ -23,15 +32,27 @@ pub fn analyze<T: FieldElement>(file: ASMProgram<T>) -> Result<AnalysisASMFile<T
 
     // run analysis on virtual machines, reducing them to constrained machines
     log::debug!("Start asm analysis");
-    let file = vm::analyze(file, &mut monitor)?;
+    let file = vm::analyze(file, monitor)?;
     log::debug!("End asm analysis");
 
+    Ok(file)
+}
+
+pub fn consume_asm<T: FieldElement>(
```

Comment:
`consume` is kind of weird because it does not say what it converts it to.
What about calling this `convert_analyzed_to_pil` and the other `convert_asm_to_pil` or something?